dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -1882,6 +1888,15 @@ def dpnp_solve(a, b):
                 depends=[a_copy_ev, b_copy_ev],
             )
 
+            # TODO: Remove this w/a when MKLD-17201 is solved.
+            # Waiting for a host task executing an OneMKL LAPACK gesv call
+            # on CPU causes deadlock due to serialization of all host tasks
+            # in the queue.
+            # We need to wait for each host tasks before calling _gesv to avoid deadlock.
+            if is_cpu_device:
+                b_ht_copy_ev[i].wait()
+                ht_lapack_ev[i].wait()
+
         for i in range(batch_size):
             ht_lapack_ev[i].wait()
             b_ht_copy_ev[i].wait()
```

Comment:
I believe it would be more safe to have:
```suggestion
            if not is_cpu_device:
                ht_lapack_ev[i].wait()
                b_ht_copy_ev[i].wait()
```